install/kubernetes/cilium/templates/_helpers.tpl

```diff
@@ -36,6 +36,17 @@ backend:
 {{- end -}}
 {{- end -}}
 
+{{/*
+Return the container image based on either .image.repository or .registry/.image.name.
+*/}}
+{{- define "container.image" -}}
+{{-  if .image.repository -}}
```

Comment:
What is the long-term plan of supporting this kind of pattern? Do we intend to keep it or do we deprecate it?

If we want to keep it, then we should probably keep them (set to nil or commented out) in the`values.yaml` file. 
Even if we deprecate it, we should probably keep it in the `values.yaml` and set a release for when we will remove it.

Reply:
I see two reasons to keep it:

1. So that it is still possible to override `repository` on a per-image basis (e.g. `--set hubble.relay.image.repository=quay.io/kaworu/hubble-relay-next`).
2. We currently use `repository` for `hubble.proxy.image.repository=docker.io/envoyproxy/envoy` and `clustermesh.etcd.image.repository=quay.io/coreos/etcd`.

Having them commented out is tricky because of how the `update-versions` make target `sed` dance is performed to override the versions (they use `name` and `repository` as anchor to find the `tag:`), but I'll take a stab at it.

Reply:
Seems to work, I tried my best to "strengthen" the sed stuff to ignore commented lines.